File: countminsketch.py
d1={0:0,1:0,2:0,3:0,4:0,5:0,6:0}
d2={0:0,1:0,2:0,3:0,4:0,5:0,6:0}
d3={0:0,1:0,2:0,3:0,4:0,5:0,6:0}

def h1(e):
    return (2*e+1)%7

def h2(e):
    return (5*e+3)%7

def h3(e):
    return (9*e+6)%7

def insert(e):
    print("hash values are : ",(h1(e),h2(e),h3(e)))
    try:
        d1[h1(e)]=d1[h1(e)]+1
    except:
        d1[h1(e)]=1
    try:
        d2[h2(e)]=d2[h2(e)]+1
    except:
        d2[h2(e)]=1
    try:
        d3[h3(e)]=d3[h3(e)]+1
    except:
        d3[h3(e)]=1
    print("INSERTED\n")

def search(e):
    a=d1[h1(e)]
    b=d2[h2(e)]
    c=d3[h3(e)]
    print("hash values are : ",(h1(e),h2(e),h3(e)))
    print("the values in cms are : ",(a,b,c))
    print("count of ",e," is  : ", min([a,b,c]))

choice=0
while True:
    if choice==0:
        e=int(input("enter element to insert : "))
        insert(e)
        choice=-1
    elif choice==-1:
        print("******************* 0=insert\t3=count\t 1=exit\t 2=print************************")
        choice=int(input("enter choice : "))
    elif choice==3:
        e=int(input("enter element to count : "))
        search(e)
        choice=-1
    elif choice==1:
        break
    elif choice==2:
        print(d1)
        print(d2)
        print(d3)
        choice=-1

File: prac/countminsketch.java
import java.io.*;
import java.util.*;

public class countminsketch{
    public static int h1(int ele){
        return (5*ele+2)%7;
    }
    public static int h2(int ele){
        return (89*ele+27)%7;
    }
    public static void main(String args[]){
        Scanner sc =new Scanner(System.in);
        Hashtable<Integer, Integer> ht1 = new Hashtable<>(7);
        Hashtable<Integer, Integer> ht2 = new Hashtable<>(7);
        for(int i=0;i<7;i++){
            ht1.put(i,0);
            ht2.put(i,0);
            }
            
        int choice=1;
        while(true){
            if(choice==1)
            {
                System.out.print("Enter element to insert : ");
                int ele=sc.nextInt();
                int a=h1(ele);
                int b=h2(ele);
                System.out.println("hash values are : "+a+ " ," +b);
                ht1.put(a,ht1.get(a)+1);
                ht2.put(b,ht2.get(b)+1);
                System.out.println("element inserted \n");
                choice=-1;
            }
            else if(choice==2){
                System.out.println(ht1);
                System.out.println(ht2);
                choice=-1;
            }
            else if(choice ==-1){
                System.out.println("*****1==insert\t 2==print ht\t4==count\t 3==exit*****");
                System.out.print("enter choice : ");
                choice=sc.nextInt();
            }
            else if(choice==3){
                break ;
            }
            else if(choice==4){
                System.out.print("Enter element to count : ");
                int ele=sc.nextInt();
                int a=h1(ele);
                int b=h2(ele);
                System.out.println("hash values are : "+a+ " ," +b);
                System.out.println("Hashtable values at above hash values are : "+ht1.get(a)+ " ," +ht2.get(b));
                System.out.println("count of  : "+ele+ " is : " + Math.min(ht1.get(a),ht2.get(b)));
                choice=-1;
            }
            
        }
        
    }

}